### PR TITLE
vm-repair: Bug Fix when HyperV-version Variable is None

### PR DESCRIPTION
**Pull request by swbae31**
Quick few liner bug fix when the HyperV-version variable is none

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S168

**Comment by swbae31** [comment]
@yonzhan 
I got an error from merge validation that 
"Once your PR is merged into master branch, a new PR will be created to update src/index.json automatically."

Has some process changed for PRs?

**Comment by yonzhan** [comment]
Yes, the process is changed to automatically release extensions. @haroldrandom can help with it.

**Comment by yonzhan** [comment]
You should not change source code and index.json at the same time right now.

**Comment by haroldrandom** [comment]
src/index.json should't modify both in one PR, we have a release pipeline for publishing your extension automatically. 
If you want to use it and hold extension wheel on CLI's official container, just leave src/index.json untouched. After this PR merge, another RP to update src/index.json will be created automatically.
If you still want to hold it in your own container, just fire another PR to update the src/index.json separately.

The detail should be listed in PR template.

File: src/vm-repair/azext_vm_repair/custom.py [hunk 1]
```diff
@@ -85,8 +85,11 @@ def create(cmd, vm_name, resource_group_name, repair_password=None, repair_usern
 
             # Copy OS disk command
             disk_sku, location, os_type, hyperV_generation = _fetch_disk_info(resource_group_name, target_disk_name)
-            copy_disk_command = 'az disk create -g {g} -n {n} --source {s} --sku {sku} --location {loc} --os-type {os_type} --hyper-v-generation {hyperV} --query id -o tsv' \
-                                .format(g=resource_group_name, n=copy_disk_name, s=target_disk_name, sku=disk_sku, loc=location, os_type=os_type, hyperV=hyperV_generation)
+            copy_disk_command = 'az disk create -g {g} -n {n} --source {s} --sku {sku} --location {loc} --os-type {os_type} --query id -o tsv' \
+                                .format(g=resource_group_name, n=copy_disk_name, s=target_disk_name, sku=disk_sku, loc=location, os_type=os_type)
+            # Only add hyperV variable when available
+            if hyperV_generation:
+                copy_disk_command += ' --hyper-v-generation {hyperV}'.format(hyperV=hyperV_generation)
```
Comment (qwordy) [inline src/vm-repair/azext_vm_repair/custom.py:92]: Looks good to me.

**Comment by swbae31** [comment]
Got it. That's great. Creating the wheel file, uploading it, and making the hash has always been really tedious and made PR changes difficult. This is an awesome change. Thanks.

Then when does the release pipeline happen? And how would I know it has been released with the index.json updated?

**Comment by yungezz** [comment]
hi @qwordy could  you pls help to review/merge this PR? thanks

**Comment by haroldrandom** [comment]
@swbae31 
> Then when does the release pipeline happen?
 It's triggered once PR is merged into master branch but only build and upload only when extension upgrade (detecting via version).

Normally PR developer won't get notifed for the PR to update src/index.json, @fengzhou-msft and I will get notification once because code owner review mechanism. We will assign it to the assignee who responsible for the source PR to trigger this one, then approve it. 
You can have a look how the other extension works. :)

**Pull request merged**